pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -300,18 +300,13 @@ func (f *Fit) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldOb
 	return framework.Queue, nil
 }
 
-// isResourceScaleDown checks whether the resource request of the modified pod is less than the original pod
-// for the resources requested by the pod we are trying to schedule.
-func (f *Fit) isResourceScaleDown(targetPod, originalOtherPod, modifiedOtherPod *v1.Pod) bool {
-	if modifiedOtherPod.Spec.NodeName == "" {
-		// no resource is freed up whatever the pod is modified.
-		return false
-	}
```

Comment:
(maybe we had discussion before and I forgot)

Why do we remove this? Instead, shouldn't it be:

```go
	if modifiedOtherPod.Spec.NodeName == "" && 
	  targetPod.UID != modifiedPod.UID { 
		return false
	}
```

?

Reply:
If it's a unscheduled pod got updated, it will only attempt to requeue itself.
So, `modifiedOtherPod.Spec.NodeName == "" && targetPod.UID != modifiedPod.UID`, this should never happen?

Reply:
> If it's a unscheduled pod got updated, it will only attempt to requeue itself.

With the current implementation, Yes. If an unscheduled Pod is updated, QHint of Pod/Update is only used for that specific unscheduled Pod's requeueing. But, this behaviour is actually a bug based on spec; that is, not documented at all. (It's raised at: https://github.com/kubernetes/kubernetes/issues/110175#issuecomment-1140397251.) 
Actually, any custom plugins may want to requeue Pods with unscheduled Pods' events (coscheduling, as an example in the community). 
So, in the future, we have to trigger Pod/Update events with unscheduled pods' update, and hense here I believe we still should consider ` modifiedOtherPod.Spec.NodeName == "" && targetPod.UID != modifiedPod.UID` so that we make this QHint not rely on undocumented behaviour, which will be changed in the future.

Reply:
> which will be changed in the future.

In this case, I'm ok with checking UID here. As this is a simple check. I'll add it later.

Reply:
Reverted related changes in https://github.com/kubernetes/kubernetes/commit/1e9d4e3eeb0a91af7958be03ab8ba3dd7e183a20